### PR TITLE
switch to PatternFly & upgrading many dependencies

### PR DESCRIPTION
**Pull request by tommartensen**
Missing upgrades:
- `@stackrox/tailwind-config` / `tailwindcss`, because `@stackrox/tailwind-config` has conflicting dependencies and won't receive fixes (for now). Can be done in a separate PR if necessary.
- `axios` - I couldn't get it to work, it also needs to be updated in `infra-auth-lib`, but then everything broke. Next time.

Proof that the UI works (local deployment):

Screenshot 2023-10-12 at 14 45 57

Backend is tested with E2E tests.

### CODE REVIEW

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -8,8 +8,7 @@ module.exports = {
     'plugin:eslint-comments/recommended',
     'airbnb-typescript',
     'prettier',
-    'prettier/@typescript-eslint',
-    'prettier/react',
+    'plugin:prettier/recommended',
```
Comment (tommartensen) [inline ui/.eslintrc.js:11]: required after react 18 upgrade

File: ui/package.json [hunk 2]
```diff
@@ -22,47 +22,54 @@
...
+    "eslint": "^8.50.0",
+    "eslint-config-airbnb-typescript": "^17.1.0",
+    "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-eslint-comments": "^3.2.0",
-    "eslint-plugin-prettier": "^3.4.1",
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-prettier": "^5.0.0",
     "http-proxy-middleware": "^2.0.1",
-    "npm-run-all": "^4.1.5",
+    "npm-run-all2": "^6.1.1",
```
Comment (tommartensen) [inline ui/package.json:69]: old library did not receive fixes anymore, this is a fork

File: ui/src/components/AppHeaderLayout.tsx [hunk 3]
```diff
@@ -1,4 +1,11 @@
 import React, { ReactElement, ReactNode } from 'react';
+import {
```
Comment (tommartensen) [inline ui/src/components/AppHeaderLayout.tsx:2]: changes very similar to in `infra` repository...

File: ui/src/components/RHACSLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import React, { forwardRef } from 'react';
```
Comment (tommartensen) [inline ui/src/components/RHACSLogo.tsx:1]: like in `infra` repository...

File: ui/src/containers/AppHeader/AppHeader.test.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ test('renders app with the proper header', () => {
   const { getByText } = render(
     <Router>
       <AppHeader />
-    </Router>
+    </Router>,
```
Comment (tommartensen) [inline ui/src/containers/AppHeader/AppHeader.test.tsx:11]: a `prettier` linter finding

File: ui/src/containers/AppHeader/ProductLogoTile.tsx [hunk 6]
```diff
@@ -1,17 +1,22 @@
...
-import StackRoxLogo from 'components//StackRoxLogo';
+import RHACSLogo from 'components/RHACSLogo';
 
 export default function ProductLogoTile(): ReactElement {
   return (
-    <div className="flex flex-col items-center pb-1 px-4 border-r border-base-400">
-      <Link to="/" className="flex items-center">
-        <StackRoxLogo />
-        <div className="pl-1 pt-1 text-sm tracking-wide font-600 font-condensed uppercase">
+    <Flex alignItems={{ default: 'alignItemsCenter' }}>
```
Comment (tommartensen) [inline ui/src/containers/AppHeader/ProductLogoTile.tsx:9]: like in infra repository...

File: ui/src/containers/AppHeader/UserInfo.tsx [hunk 7]
```diff
@@ -1,18 +1,24 @@
 import React, { ReactElement } from 'react';
 
-import { Avatar } from '@stackrox/ui-components';
+import { Avatar } from '@patternfly/react-core';
```
Comment (tommartensen) [inline ui/src/containers/AppHeader/UserInfo.tsx:3]: like in infra repository...

File: ui/src/containers/HomePage/UserServiceAccountToken.tsx [hunk 8]
```diff
@@ -30,7 +30,7 @@ export default function UserServiceAccountToken(): ReactElement {
 
   if (error || !data?.Token) {
     return (
-      <Tooltip content={<TooltipOverlay>{error?.message || 'Unknown error'}</TooltipOverlay>}>
+      <Tooltip content={<div>{error?.message || 'Unknown error'}</div>}>
```
Comment (tommartensen) [inline ui/src/containers/HomePage/UserServiceAccountToken.tsx:33]: like in infra repository...

File: ui/src/index.tsx [hunk 9]
```diff
@@ -1,12 +1,15 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
```
Comment (tommartensen) [inline ui/src/index.tsx:2]: Required change after `react-dom` upgrade

**Review by gavin-stackrox — APPROVED** [review]
:yay!:

**Pull request merged**